### PR TITLE
Return data frames from `vec_data()`

### PR DESCRIPTION
**Pull request by lionel-**
Shouldn't affect dplyr as it uses `dplyr_vec_data()` which already preserves data frames.

### CODE REVIEW

File: R/proxy.R [hunk 1]
```diff
@@ -207,6 +207,10 @@ vec_data <- function(x) {
   vec_assert(x)
   x <- vec_proxy(x)
 
+  if (is.data.frame(x)) {
+    return(new_data_frame(x, row.names = attr(x, "row.names")))
```
Comment (DavisVaughan) [inline R/proxy.R:211]: Does this have performance implications? Doesn't this materialize compact row names?

Comment (lionel-) [reply R/proxy.R:211]: oops you're right, `attr()` doesn't return a character vector unlike `row.names()`, but it does materialise. We should probably use `.row_names_info(x, 0)` here.

Comment (DavisVaughan) [reply R/proxy.R:211]: we have used `.row_names_info(data, type = 0L)` many times in dplyr. is it worth making an rlang helper for that?

Comment (lionel-) [reply R/proxy.R]: Now done, thanks!

**Pull request merged**